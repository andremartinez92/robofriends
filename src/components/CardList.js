// @flow

import React from 'react';

import { type RobotData } from '../types';

import Card from './Card';

type Props = {
  robots: RobotData[],
};

function CardList(props: Props) {
  const { robots } = props;

  return (
    <React.Fragment>
      {robots.map(robot => (
        <Card
          key={robot.id}
          email={robot.email}
          id={robot.id}
          name={robot.name}
        />
      ))}
    </React.Fragment>
  );
}

export default CardList;
