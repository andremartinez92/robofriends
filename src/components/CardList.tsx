import React from 'react';

import { RobotData } from '../types';

import Card from './Card';

interface Props {
  robots: RobotData[];
}

function CardList(props: Props): JSX.Element {
  const { robots } = props;

  return (
    <>
      {robots.map((robot: RobotData) => (
        <Card
          key={robot.id}
          email={robot.email}
          id={robot.id}
          name={robot.name}
        />
      ))}
    </>
  );
}

export default CardList;
