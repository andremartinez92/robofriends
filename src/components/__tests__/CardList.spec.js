import React from 'react';
import { shallow } from 'enzyme';

import CardList from '../CardList';

describe('<CardList>', () => {
  const props = {
    robots: [
      {
        email: 'test@test.com',
        id: 12,
        name: 'some name',
      },
      {
        email: 'dev@dev.com',
        id: 34,
        name: 'dev robot',
      },
    ],
  };

  it('renders the expected component', () => {
    const subject = shallow(<CardList {...props} />);
    expect(subject).toMatchSnapshot();
  });
});
