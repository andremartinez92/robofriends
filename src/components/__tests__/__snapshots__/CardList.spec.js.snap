// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList> renders the expected component 1`] = `
<Fragment>
  <Card
    email="test@test.com"
    id={12}
    key="12"
    name="some name"
  />
  <Card
    email="dev@dev.com"
    id={34}
    key="34"
    name="dev robot"
  />
</Fragment>
`;
